auto_tune_vllm/core/config.py

```
@@ -518,15 +529,44 @@ def _validate_config(self, raw_config: Dict[str, Any]) -> StudyConfig:
+            # Check if this is an environment variable (has explicit type or not in schema)
+            param_type = param_config.get("type")
             schema_def = self.schema.get("parameters", {}).get(param_name)
-            if not schema_def:
-                raise ValueError(f"Unknown parameter in schema: {param_name}")
             
-            # Build parameter config based on type
-            validated_param = self._build_parameter_config(param_name, param_config, schema_def)
+            if param_type == "environment" or (not schema_def and "options" in param_config):
+                # This is an environment variable parameter
+                if "range" in param_config or "min" in param_config or "max" in param_config or "step" in param_config:
+                    raise ValueError(f"Environment parameter '{param_name}' cannot use range configurations. Only list options are allowed.")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)